Release 2.1

Underdog-01 opened the pull request:
Additional commands for package-info.xml (installing/uninstalling mods)

Added additional commands for installing/uninstalling modifications for SMF.
The edits give mod authors the ability to add various input types (including textarea) to the install/uninstall packages template. These inputs can be processed in any of their installation files used from xml code tags.

Pending: < script > (Javascript) xml tags to execute javascript code for the packages template and also javascript input/textarea attributes to be added to this current addition (which can possibly use the added javascript as functions).

Syntax and examples are located in the additional text README file in my SMF 2.1 repository.

Commit:
Update Subs-Package.php

Source code to accept <input> tags from package-info.xml file.

Syntax and examples are noted on my main fork for SMF2.1 repository.
Other suggestions:
- add ability to accept <javascript> tags from package-info.xml
- add javascript options for input 

This will give mod authors more freedom of db manipulation during install and/or uninstall.

Commit:
Update Packages.php

This fills the $context['new_inputs'] array with the necessary $_POST data from the form.

Commit:
Update Packages.template.php

If there were inputs fed into the $context['new_package_inputs'] array from Subs-Package.php, this will display them.

Commit:
Create Underdog-Additions-README..txt

Explanation of changes made.

Commit:
Update Subs-Package.php

Added syntax for class attributes.

Commit:
Update Underdog-Additions-README..txt

Commit:
Update Underdog-Additions-README..txt

Commit:
Update Underdog-Additions-README..txt

emanuele45 commented:
Please note that in order to merge that the commits must be signed-off as per [contribution instructions](https://github.com/SimpleMachines/SMF2.1#how-to-contribute).
Thanks. ;)

It is possible to rebase the branch and automatically sign-off the commits, at the moment I can't find the page where the procedure is described...

Commit:
Update Subs-Package.php

condensed one condition

Commit:
Update Underdog-Additions-README..txt

style elements behaved somewhat different on SMF 2.1 Alpha as opposed to SMF 2.0.4
Changed. 
Example looks fine and works correctly for SMF 2.1 Alpha.
